MetaMask/metamask-extension
# OpenSea security provider metrics

## Explanation
* Fixes #16262 
* Fixes #18150

Added metrics for the `OpenSea` security provider. 

Created a new user profile property called `security_provider`. This property is a type list and receives the value `opensea` if the user has enabled the `OpenSea` security provider feature under `Settings > Experimental`.
If the user turns off the security provider under `Settings > Experimental`, then the user profile property `security_provider` is updated and the `opensea` value removed from it.

Additionally there is a new property called `ui_customizations` added to `Transaction Approved`, `Transaction Rejected`, `Signature Approved` and `Signature Rejected` events.
This property receives one of the following values: 

1. `['flagged_as_malicious']` when the transaction was flagged as malicious
2. `['flagged_as_safety_unknown']` when the transaction was flagged as safety unknown
3. `null` in all other cases 

by the transaction security provider.

## Manual testing steps
In the `.metamaskrc` set the transaction security feature flag to 1 - `TRANSACTION_SECURITY_PROVIDER=1`.

Follow the procedure for testing the metrics event and check the response in the background console of the extension - `background.html`.

Cases to check:
- enable/disable the `OpenSea` security provider under `Settings > Experimental`
- `SEND LEGACY TRANSACTION`
- `SEND EIP 1559 TRANSACTION`
- `ETH SIGN`
- `PERSONAL SIGN`
- `SIGN TYPED DATA`
- `SIGN TYPED DATA V3`
- `SIGN TYPED DATA V4`
